### PR TITLE
Adding hookup and setup instructions for NVIDIA Jetson

### PR DESCRIPTION
**Pull request by tokk-nv**
Hi,

I added hardware hookup and software setup instructions for NVIDIA Jetson.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,18 +8,25 @@ The automatically generated website can be viewed at https://iroboteducation.git
 Install required dependencies:
 
 ```bash
+sudo apt-get update
+sudo apt-get -y install python3-pip
+sudo apt-get -y install mkdocs
```
Comment (alsora) [inline README.md:13]: No need for this as it's equivalent to the line below.

File: docs/hw/hookup.md [hunk 2]
```diff
@@ -31,4 +31,17 @@ The Jetson Nano™ 2GB has a USB-C® port (J2) for power and a USB Micro-B port
...
+The [Jetson Xavier NX Developer Kit](https://developer.nvidia.com/embedded/jetson-xavier-nx-devkit) has a 5.5mm x 2.5mm barrel connector jack (`J16`) for power (9V to 20V) and a USB Micro-B port (`J5`) for downstream data.<br>
+This can be powered from the unregulated battery port of the Create® 3  adapter board by using a JST-XH female connector to DC barrel plug cable.
+The data connection is established by using a USB Micro B to USB-C® cable.
+![Hookup diagram for Jetson Xavier NX Developer Kit](data/hookup_jetson_xavier_nx.jpg "Jetson Nano™ Xavier NX Developer Kit")
+
+### Mouting NVIDIA® Jetson Xavier NX Developer Kit
+You can 3d print the [mount adapter](data/C3-JetsonXavierNX-Mount.3mf) to place the Jetson Xavier NX Developer Kit in the cargo bay or on the faceplate.
+
+!!! note
+    - If you are 3d printing the above mount adapter, use **support** to surport overhang areas. ([slicing example](data/C3-JetsonXavierNX-Mount_slice-example.png))
```
Comment (alsora) [inline docs/hw/hookup.md:44]: Typo "surport"

File: docs/setup/jetson.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+1. Download the SD card image for your Jetson platform and write onto a microSD card.
+
+2. Complete the getting started guide (listed above) until you complete the initial setup.
+
+### Alter "USB Device Mode" to apply static IP 
+
+1. To alter the Jetson's "USB Device Mode" feature (specifically, disabling DHCP server and self-assign a static IP address);
+
+    1. First, create a flag file to store IP address for Jetson to be used with Create3.
```
Comment (alsora) [inline docs/setup/jetson.md:41]: Create3 -> Create® 3
This applies to all occurrences.

File: docs/setup/jetson.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+!!! note
+    If you ever want to stop using the Jetson device for Create3 and re-enable the original USB Device Mode feature (so that you can connect to your Windows PC in headless style), you can simply remove the flag file, or execute the following.
+
+        sudo mv /opt/nvidia/l4t-usb-device-mode/IP_ADDRESS_FOR_CREATE3_ROBOT.conf /opt/nvidia/l4t-usb-device-mode/IP_ADDRESS_FOR_CREATE3_ROBOT.conf.bak
+
+
+### Start ROS 2 Galactic container
+
+The recommended way to run ROS2 on Jetson is to use a pre-built Docker containers.
```
Comment (alsora) [inline docs/setup/jetson.md:71]: Is the docker engine already included in the Jetson image provided above?

File: docs/setup/jetson.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+1. Enable NVIDIA runtime for docker, if this is the first time running Docker containers on Jetson, to allow access to GPU from containers.
+
+    Edit `/etc/docker/daemon.json` to be following.
```
Comment (alsora) [inline docs/setup/jetson.md:75]: "to be THE following" or "to be AS FOLLOWS"

File: README.md [hunk 6]
```diff
@@ -8,18 +8,25 @@ The automatically generated website can be viewed at https://iroboteducation.git
...
 pip3 install mkdocs
 pip3 install mkdocs-material
 ```
 
 Build the documentation locally:
 
 ```bash
-mkdocs serve
+python3 -m mkdocs serve
```
Comment (shamlian) [inline README.md:21]: Something seems not right here -- if you install mkdocs with apt, you shouldn't need to invoke it in this way,

File: mkdocs.yml [hunk 7]
```diff
@@ -29,6 +29,7 @@ nav:
       - Connect to Wi-Fi: setup/provision.md
       - Multi-Robot Setup: setup/multi-robot.md
       - ROS 2 on a Pi 4: setup/pi4.md
+      - ROS 2 on a NVIDIA Jetson: setup/jetson.md
     - Create® 3 APIs:
```
Comment (shamlian) [inline mkdocs.yml:33]: We may have a merge conflict soon if iRobot's branding PR gets merged before we resolve all the issues in this review. Just a heads up.

File: docs/setup/jetson.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
+<!--
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (shamlian) [inline docs/setup/jetson.md:4]: Is there a reason that using BSD 3-clause as with the rest of this repository is not acceptable? If NVIDIA's contributions must be Apache 2.0, then we will also need to update LICENSE.txt with exceptions, and that will need to be kept up to date. I would prefer if possible to keep the same license for the whole project.

**Comment by shamlian** [comment]
May want to wait to merge this until after the NVIDIA PR comes in.

File: README.md (hunk 1 above)
Comment (tokk-nv) [reply README.md]: Reverted this change in commit 8259333

File: README.md (hunk 6 above)
Comment (tokk-nv) [reply README.md]: Reverted this change in commit 8259333

File: docs/setup/jetson.md (hunk 5 above)
Comment (tokk-nv) [reply docs/setup/jetson.md]: Fixed in commit 5a44eb4

File: docs/setup/jetson.md (hunk 4 above)
Comment (tokk-nv) [reply docs/setup/jetson.md]: Yes

File: docs/setup/jetson.md (hunk 8 above)
Comment (tokk-nv) [reply docs/setup/jetson.md]: Copyright notice changed to BSD-3.
Resolved in commit 2cfff17

File: docs/setup/jetson.md (hunk 3 above)
Comment (tokk-nv) [reply docs/setup/jetson.md:56]: Fixed in commit 363a5df

File: docs/hw/hookup.md (hunk 2 above)
Comment (tokk-nv) [reply docs/hw/hookup.md]: Fixed in commit 363a5df

**Review by shamlian — APPROVED** [review]

**Pull request merged**